src/index.js

```diff
@@ -55,6 +64,22 @@ export default class LazilyRender extends React.Component<LazilyRenderProps, Laz
     return convertOffsetToBounds(offset);
   }
 
+  componentDidUpdate(prevProps: ?LazilyRenderProps) {
+    const { scrollContainer: prevContainer } = prevProps;
+    const { scrollContainer: container } = this.props;
+    // If a scroll container was defined before, clear its scroll event listener.
+    if (prevContainer && (prevContainer !== container)) {
+      prevContainer.removeEventListener('scroll', this.update, eventListenerOptions);
```

Comment:
I think we should reuse `stopListening` here instead

Reply:
I wanted to avoid using `stopListening` and `startListening` as they introduce a new listener on `window` which we probably don't want (?). I can tease the logic out of `startListening` and `stopListening` but they'll turn into one-liners then - wasn't sure if that was a great idea 😄

Reply:
Gotcha - will do that 👍